include/zephyr/sys/util.h

```diff
@@ -259,6 +260,26 @@ extern "C" {
  */
 #define DIV_ROUND_UP(n, d) (((n) + (d) - 1) / (d))
 
+/**
+ * @brief Divide and round to the nearest integer.
+ *
+ * Example:
+ * @code{.c}
+ * DIV_ROUND_CLOSEST(5, 2); // 3
+ * DIV_ROUND_CLOSEST(5, -2); // -3
+ * DIV_ROUND_CLOSEST(5, 3); // 2
+ * @endcode
+ *
+ * @param n Numerator.
+ * @param d Denominator.
+ *
+ * @return The result of @p n / @p d, rounded to the nearest integer.
+ */
+static ALWAYS_INLINE int DIV_ROUND_CLOSEST(int n, int d)
```

Comment:
UPPERCASE is reserved for macros, no?

Reply:
(note: neither Linux nor this approach allow for compile time initialization).

Reply:
Right, but at the same time we might want to keep it this way to be "compatible" with Linux, see https://elixir.bootlin.com/linux/latest/source/include/linux/math.h#L86.

Honestly I don't feel too strong about this, I can change it back to lowercase if you want me to.

Reply:
Yeah also neither Linux or us are very coherent with the macro uppercase naming and coherency, see `container_of`... I'd say make it a macro, we can live with the double evaluation for now since we have it in the other one, then we can rework it down the road with the Linux `typeof` trick, which you could not do if it was an inline.

Reply:
Converted it back to a macro.
There is one potentially problematic thing about it though.
In the new version that checks for negative integers  "n" is evaluated 3 times, and "d" 5 times.
I'm don't think that's a problem since the compiler should be able to optimize that anyway.

Reply:
If we decide to fix them we can fix them in bulk, at least they'd be coherent.

Reply:
Whee, macrobatics.

> UPPERCASE is reserved for macros, no?

FWIW, the standard workaround to that sort of thing is to define a __internal_function() that then gets expanded from EXTERNAL_MACRO().

Regarding compile time evaluation: that's true, and (until `constexpr` is universally available) something that has to be sort of a tradeoff between macro hygiene and runtime performance.   For "round to nearest" I don't know what the right choice is, but I'd certainly agree that power-of-2 alignment/padding macros want to be statically optimizable.